slogr/slogr.go

```diff
@@ -105,3 +105,24 @@ type SlogSink interface {
 	WithAttrs(attrs []slog.Attr) SlogSink
 	WithGroup(name string) SlogSink
 }
+
+// HandlerFromContext retrieves a slog.Handler from the context or, if no
+// logger is stored there, returns the slog default handler.
+//
+// It uses logr.FromContext and thus is interoperable with code that only knows
+// about the logr API.
+func HandlerFromContext(ctx context.Context) slog.Handler {
+	logger, err := logr.FromContext(ctx)
+	if err == nil {
+		return NewSlogHandler(logger)
+	}
+	return slog.Default().Handler()
+}
+
+// ContextWithHandler creates a new context which contains the given handler.
+//
+// It stores the handler after conversion to a logr.Logger with logr.NewContext
+// and thus is interoperable with code that only knows about the logr API.
+func ContextWithHandler(ctx context.Context, handler slog.Handler) context.Context {
```

Comment:
Should this be `NewContext` for consistency?

Reply:
I wanted `Handler` in the name because a variant with `*slog.Logger` as parameter might also be useful at some point.

If we used `slogr.NewContext`, it would probably be better defined as taking `*slog.Logger` as parameter for full consistency with `logr.NewContext`.

As it stands now, the name is more consistent with `context.WithDeadline` or `context.WithCancel`. We cannot have consistency with both the `logr` and `context` package, though, and consistency with `logr` is more important ("closer" package).

`NewContextWithHandler` (and future `NewContextWithLogger`)?